core/unit_test/tstNeighborList.hpp

```diff
@@ -553,6 +553,79 @@ void testVerletListFullPartialRange()
                                        num_ignore );
 }
 
+//---------------------------------------------------------------------------//
+template <class LayoutTag>
+void testAngularParallelFor()
+{
+    // Create the AoSoA and fill with random particle positions.
+    int num_particle = 1e3;
+    double test_radius = 2.32;
+    double cell_size_ratio = 0.5;
+    double box_min = -5.3 * test_radius;
+    double box_max = 4.7 * test_radius;
+    auto aosoa = createParticles( num_particle, box_min, box_max );
+
+    // Create the neighbor list.
+    using ListType =
+        Cabana::VerletList<TEST_MEMSPACE, Cabana::FullNeighborTag, LayoutTag>;
+    double grid_min[3] = {box_min, box_min, box_min};
+    double grid_max[3] = {box_max, box_max, box_max};
+    ListType nlist( Cabana::slice<0>( aosoa ), 0, aosoa.size(), test_radius,
+                    cell_size_ratio, grid_min, grid_max );
+
+    // Create Kokkos views for the write operation.
+    using memory_space = typename TEST_MEMSPACE::memory_space;
+    Kokkos::View<int *, Kokkos::HostSpace> test_result( "test_result",
+                                                        num_particle );
+    Kokkos::View<int *, memory_space> serial_result( "serial_result",
+                                                     num_particle );
+    Kokkos::View<int *, memory_space> team_result( "team_result",
+                                                   num_particle );
+
+    // Test the list parallel operation by adding a value from each neighbor
+    // to the particle and compare to counts.
+    auto serial_count_op =
+        KOKKOS_LAMBDA( const int i, const int j, const int k )
+    {
+        Kokkos::atomic_add( &serial_result( i ), j );
+        Kokkos::atomic_add( &serial_result( i ), k );
+    };
+    auto team_count_op = KOKKOS_LAMBDA( const int i, const int j, const int k )
+    {
+        Kokkos::atomic_add( &team_result( i ), j );
+        Kokkos::atomic_add( &team_result( i ), k );
+    };
+    Kokkos::RangePolicy<TEST_EXECSPACE> policy( 0, aosoa.size() );
+    Cabana::angular_neighbor_parallel_for( policy, serial_count_op, nlist,
```

Comment:
instead of a new function I wonder if we could generalize the `neighbor_parallel_for` with something like a neighbor depth parameter?

Reply:
Agreed, I renamed all to `neighbor_parallel_for`, where the angular variations have two parallelism tags instead of one